packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -252,14 +259,63 @@ export class EntityService {
     });
   }
 
+  async checkServiceFeatureUnderLimitation(
+    workspaceId: string,
+    resourceId: string,
+    serviceFeature: EnumServiceFeature
+  ): Promise<void> {
+    const feature = await this.billingService.getMeteredEntitlement(
+      workspaceId,
+      BillingFeature.Services
+    );
+
+    if (feature.hasAccess) {
+      return;
+    }
+
+    const featuredEntityOrderedByCreationDate =
+      await this.prisma.resource.findMany({
+        where: {
+          deletedAt: null,
+          archived: { not: true },
+          resourceType: EnumResourceType.Service,
+          project: {
+            workspaceId,
+            deletedAt: null,
+          },
+        },
+        orderBy: {
+          createdAt: "asc",
+        },
+        skip: feature.usageLimit,
+      });
+
+    const canCreate = featuredEntityOrderedByCreationDate.some(
+      (featureEntity) => featureEntity.id === resourceId
+    );
+
+    if (!canCreate) {
+      throw new AmplicationError(
```

Comment:
Let's use the new `BillingLimitationError` instead of the generic AmplicationError.